pkg/controller/networkpolicy/networkpolicy_controller_perf_test.go

```diff
@@ -85,56 +168,179 @@ func getXLargeScaleWithLargeNamespaces() ([]*corev1.Namespace, []*networkingv1.N
+	namespace := rand.String(8)
+	getObjects := func() ([]*corev1.Namespace, []*v1alpha1.ClusterNetworkPolicy, []*corev1.Pod) {
+		label := map[string]string{"app-1": "scale-1"}
+		namespaces := []*corev1.Namespace{newNamespace(namespace, map[string]string{"app": namespace})}
+		acnps := []*v1alpha1.ClusterNetworkPolicy{newClusterNetworkPolicy("", "", 0, newCRDsPeer(label, nil), newCRDsPeer(label, nil), nil, &allowAction, nil)}
+		pods := []*corev1.Pod{newPod(namespace, "", label)}
+		return namespaces, acnps, pods
+	}
+	namespaces, acnps, pods := getXObjectsACNP(10000, getObjects)
+	return namespaces[0:1], acnps, pods
+}
+
+/*
+The reference value is:
+NAMESPACES   PODS    K8s-NETWORK-POLICIES   CLUSTER-NETWORK-POLICIES   ANTREA-NETWORK-POLICES   TIME(s)    MEMORY(M)    EXECUTIONS    EVENTS(ag, atg, np)
+1            10000   0                      0                          10000                    0.31       105          10001         10001 2 10001
+*/
+func TestInitXLargeScaleWithOneNamespaceANP(t *testing.T) {
+	namespaces, anps, pods := getXLargeScaleWithOneNamespaceANP()
+	testComputeNetworkPolicy(t, 15*time.Second, namespaces, nil, nil, anps, pods)
+}
+
+func getXLargeScaleWithOneNamespaceANP() ([]*corev1.Namespace, []*v1alpha1.NetworkPolicy, []*corev1.Pod) {
+	namespace := rand.String(8)
+	getObjects := func() ([]*corev1.Namespace, []*v1alpha1.NetworkPolicy, []*corev1.Pod) {
+		label := map[string]string{"app-1": "scale-1"}
+		namespaces := []*corev1.Namespace{newNamespace(namespace, map[string]string{"app": namespace})}
+		anps := []*v1alpha1.NetworkPolicy{newAntreaNetworkPolicy(namespace, "", "", 0, newCRDsPeer(label, nil), newCRDsPeer(label, nil), nil, &allowAction, nil)}
+		pods := []*corev1.Pod{newPod(namespace, "", label)}
+		return namespaces, anps, pods
+	}
+	namespaces, anps, pods := getXObjectsANP(10000, getObjects)
+	return namespaces[0:1], anps, pods
+}
+
+/*
+TestInitXLargeScaleWithNetpolPerPod* tests the execution time and the memory usage of computing a scale
+of 1 Namespace, 10k Pods, 10k NetworkPolicies. 1 NP per Pod, with one ingress rule. The reference value is:
+
+NAMESPACES   PODS    K8s-NETWORK-POLICIES   CLUSTER-NETWORK-POLICIES   ANTREA-NETWORK-POLICES   TIME(s)    MEMORY(M)    EXECUTIONS    EVENTS(ag, atg, np)
+1            10000   10000                  0                          0                        1.86       169          3281          3281 3282 3281
+1            10000   10000                  0                          0                        2.20       170          2958          2958 2959 2958
+1            10000   10000                  0                          0                        19.90      198          10001         10001 10001 10001
+1            10000   10000                  0                          0                        24.97      185          10001         10001 10001 10001
+
+The metrics are not accurate under the race detector, and will be skipped when testing with "-race".
+*/
```

Comment:
This case and `TestInitXLargeScaleWithSmallNamespaces` above are unstable. I suspect the cleanup process. Still investigating.